model-optimizer/extensions/front/tf/SSliceComplexRolledFFTPackBlockReplacement.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging as log
+
+from extensions.middle.InsertLayoutPropagationTransposes import mark_input_as_in_correct_layout
+from extensions.ops.dft import DFT, IDFT
+from extensions.ops.roll import Roll
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.subgraph_matcher import SubgraphMatch
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+
+
+class SSliceComplexRolledFFTPackBlockReplacement(FrontReplacementSubgraph):
+    enabled = True
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('strided_slice_real', dict(op='StridedSlice')),
+                ('strided_slice_imag', dict(op='StridedSlice')),
+                ('complex', dict(op='Complex')),
+                ('roll', dict(op='Roll')),
+                ('fft', dict(op='TFFFT')),
+                ('unroll', dict(op='Roll')),
+                ('real', dict(op='Real')),
+                ('imag', dict(op='Imag')),
+                ('pack', dict(op='Pack')),
+            ],
+            edges=[
+                ('strided_slice_real', 'complex', {'in': 0}),
+                ('strided_slice_imag', 'complex', {'in': 1}),
+                ('complex', 'roll', {'in': 0}),
+                ('roll', 'fft', {'in': 0}),
+                ('fft', 'unroll', {'in': 0}),
+                ('unroll', 'real', {'in': 0}),
+                ('unroll', 'imag', {'in': 0}),
+                ('real', 'pack', {'in': 0}),
+                ('imag', 'pack', {'in': 1}),
+            ])
+
```

Comment:
The pattern is very big and transformation looks very model-specific. Can it be decomposed into several smaller one so there will be higher chance that it will work for other models?

Reply:
This transformation was split into the sequence of three transformations: `SSliceComplexRollReplacement` -> `RollRealImagPackReplacement` -> `TFFFTToDFT`. Written tests for all these transformations.